_episodes/XX-access-data.md

```diff
@@ -0,0 +1,329 @@
+~~~
+{: .output}
+
+We can print a minimal description of the available assets:
+
+~~~
+for key, asset in assets.items():
+    print(f"{key}: {asset.title}")
+~~~
+{: .language-python}
+
+~~~
+thumbnail: Thumbnail
+overview: True color image
+info: Original JSON metadata
+metadata: Original XML metadata
+visual: True color image
+B01: Band 1 (coastal)
+B02: Band 2 (blue)
+B03: Band 3 (green)
+B04: Band 4 (red)
+B05: Band 5
+B06: Band 6
+B07: Band 7
+B08: Band 8 (nir)
+B8A: Band 8A
+B09: Band 9
+B11: Band 11 (swir16)
+B12: Band 12 (swir22)
+AOT: Aerosol Optical Thickness (AOT)
+WVP: Water Vapour (WVP)
+SCL: Scene Classification Map (SCL)
+~~~
+{: .output}
+
+Among the others, assets include multiple raster data files (one per optical band, as acquired by the multi-spectral
+instrument), a thumbnail, a true-color image ("visual"), instrument metadata and scene-classification information
+("SCL"). A URL links to the actual asset:
+
+~~~
+print(assets["thumbnail"].href)
+~~~
+{: .language-python}
+
+~~~
+https://roda.sentinel-hub.com/sentinel-s2-l1c/tiles/31/U/FU/2021/1/12/0/preview.jpg
+~~~
+{: .output}
+
+This can be used to download the corresponding file:
+
+<img src="../fig/XX-STAC-s2-preview.jpg" title="Scene thumbnail" alt="true color image scene preview" width="612" style="display: block; margin: auto;" />
+
+Remote raster data can also be directly opened via `rioxarray`.
+~~~
+import rioxarray
+visual_href = assets["visual"].href
+visual = rioxarray.open_rasterio(visual_href)
+print(visual)
+~~~
+{: .language-python}
+
+~~~
+<xarray.DataArray (band: 3, y: 10980, x: 10980)>
+[361681200 values with dtype=uint8]
+Coordinates:
+  * band         (band) int64 1 2 3
+  * x            (x) float64 6e+05 6e+05 6e+05 ... 7.098e+05 7.098e+05 7.098e+05
+  * y            (y) float64 5.9e+06 5.9e+06 5.9e+06 ... 5.79e+06 5.79e+06
+    spatial_ref  int64 0
+Attributes:
+    _FillValue:    0.0
+    scale_factor:  1.0
+    add_offset:    0.0
+~~~
+{: .output}
+
+As for local files, where calling `open_rasterio` does not actually load data into memory, also for remote data the only
+metadata is fetched when opening a file. The full data content is actually downloaded when required - for instance when
+performing some raster calculations, like cropping the image using a bounding box:
+~~~
+visual_clip = visual.rio.clip_box(
+    minx=627000,
+    miny=5802000,
+    maxx=631000,
+    maxy=5806000
+)
+visual_clip.plot.imshow(figsize=(10,10))
+~~~
+{: .language-python}
+
+<img src="../fig/XX-STAC-s2-true-color-image-cutout.png" title="Scene cutout true color image" alt="RGB representation of the scene cutout" width="612" style="display: block; margin: auto;" />
+
+Finally, data can be saved to disk:
+
+~~~
+# save processed image to disk
+visual_clip.rio.to_raster("amsterdam_tci.tif", driver="COG")
+~~~
+{: .language-python}
```

Comment:
this is awesome. I think we should end with a challenge so they can reproduce these steps and build some muscle memory for how interacting with a STAC API via pystac and then working with the result in rioxarray feels.

I think a good option would be to direct them to this STAC catalog and have them download data that intersects a specific lat, lon and date (specified in the challenge text): https://radiantearth.github.io/stac-browser/#/external/earth-search.aws.element84.com/v0/collections/landsat-8-l1-c1

the solution to the challenge could be to save a single band at that location and date to disk with rioxarray

Reply:
Good point! I have added a challenge using the Landsat 8 dataset. This collection unfortunately seems not to be continuously updated here (and at a certain point might be dropped?) so we might have to find new sources in future!